### PR TITLE
Add Variant image to Entity Viewer variant page

### PR DESCRIPTION
**Pull request by azangru**
## Description
Adding variant image to the main EntityViewer view for variant. Clicking on an allele in the image changes the selected allele by updating the url.

**Note:** the nuances for some variant types, such as insertions, deletions, and indels. Alternative alleles of those variants, as reported by the variation API, will have sequences that start with the "anchor base", i.e. the nucleotide from the reference sequence that is immediately to the left of the altered sequence.

For example, for insertion, this is the diagram of what the api reports (i've coloured the anchor base green):

whereas this is how we should draw this (ignoring the first nucleotide of alt allele):

The similar consideration applies to deletions and indels.

### NOTE
See also PR https://github.com/Ensembl/ensembl-client/pull/1077, whose branch has been created from the current branch, and which, as a result, has been merged into this PR

### TODO in subsequent PR(s)
- add the "View in" button
- complex scrolling pattern (keep the reference sequence with the variant while scrolling)

## Related JIRA Issue(s)
- https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2382
- https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2381

## Deployment URL(s)
http://ev-variant-image.review.ensembl.org

**NOTE:** refget on review deployments is half-broken and cannot fetch various human chromosomes. Please test locally.

**Example variants:**
- **SNV:** http://ev-variant-image.review.ensembl.org/entity-viewer/grch38/variant:8:26293803:rs1313451503
- **SNV:** variant:1:230710048:rs699
- **substitution:** variant:1:230710043:rs141900991 (refget can't fetch chrom 1 on review deployments)
- **deletion:** variant:16:89919802:rs1555623833
- **long insertion:** variant:21:45988497:rs4647797
- **indel:** variant:1:230710046:rs878890662
- **indel, 2 alleles:** variant:21:45988303:rs1413726367

### CODE REVIEW

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+type Props = {
+  regionSliceStart: number;
+  variantStart: number; // accounts for anchor base in appropriate variant types
+  variantLength: number; // length of reference allele; accounts for anchor base in appropriate variant types
+  alleleType: string;
+  sequence: string;
+  mostSevereConsequence: string;
+  hasAnchorBase: boolean;
+  // onClick: () => void;
```
Comment (imransl) [inline src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx:33]: Remove comment?

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    : defaultLetterColour;
+
+  const sequenceLetterStyle = {
+    ['--sequence-block-color' as string]: letterColour
+  };
+  // if the sequence should end in ellipsis, make the letter block transparent
+  const lastSequenceLetterStyle = shouldShowEllipsis
+    ? {
+        ['--sequence-block-color' as string]: 'transparent',
+        ['--sequence-block-letter-color' as string]: letterColour
```
Comment (imransl) [inline src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx:82]: Is it going to be `colour` or `color` in TypeScript?

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-alleles/AlternativeAlleles.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+import styles from './AlternativeAlleles.module.css';
+
+type Props = {
+  variant: VariantDetails;
+  regionSliceStart: number;
+  variantStart: number; // accounts for anchor base in appropriate variant types
+  variantLength: number; // accounts for anchor base in appropriate variant types
+  hasAnchorBase: boolean;
+  // onClick: () => void;
```
Comment (imransl) [inline src/content/app/entity-viewer/variant-view/variant-image/alternative-alleles/AlternativeAlleles.tsx:33]: Remove comment?

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx (hunk 2 above)
Comment (azangru) [reply src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx:82]: 🤷‍♂️ 
I know that we are using `color` in CSS; so I expect the CSS variable `'--sequence-block-letter-color'` to be without the `u`. As for the name of the `letterColour` variable, I would personally love to switch to `color`; but this is not a very strong opinion.

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx (hunk 1 above)
Comment (azangru) [reply src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx:33]: This onClick comment and the one you pointed out below is a placeholder for the real click handler coming in a subsequent PR. AlternativeAllele will to respond to clicks.

File: src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+          key={index}
+          letter={letter}
+          className={letterBlockClasses}
+          style={sequenceLetterStyle}
+        />
+      ));
+    const twoDotBlocks = [...Array(2)].map((_, index) => (
+      <SequenceLetterBlock
+        key={index}
+        letter={'.'}
```
Comment (imransl) [inline src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx:115]: Can't this be as a normal HTML property value?

File: src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        className={letterBlockClasses}
+      />
+    ));
+    const missingSequenceLength = sequence.length - 20;
```
Comment (imransl) [inline src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx:119]: Add a comment to describe why sequence length is subtracted by 20?

File: src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx (hunk 1 above)
Comment (azangru) [reply src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx]: I've eventually added the onClick behaviour (clicking on an allele makes it active).

File: src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx (hunk 4 above)
Comment (azangru) [reply src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx]: ✅

File: src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx (hunk 5 above)
Comment (azangru) [reply src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx]: ✅

**Review by imransl — APPROVED** [review]

**Pull request merged**